AnkiDroid/src/main/java/com/ichi2/libanki/Decks.java

```diff
@@ -539,29 +537,28 @@ private boolean _isParent(String parentDeckName, String childDeckName) {
 
 
     private boolean _isAncestor(String ancestorDeckName, String descendantDeckName) {
-        List<String> ancestorDeckPath = path(ancestorDeckName);
-        List<String> descendantDeckPath = path(descendantDeckName);
-        if (ancestorDeckPath.size() > descendantDeckName.size()) {
+        String[] ancestorDeckPath = path(ancestorDeckName);
+        String[] descendantDeckPath = path(descendantDeckName);
+        if (ancestorDeckPath.length > descendantDeckPath.length) {
             return false;
         }
-        Iterator<String> apIt = ancestorDeckPath.iterator();
-        Iterator<String> dpIt = descendantDeckPath.iterator();
-        while (apIt.hasNext() && dpIt.hasNext()) {
-            if (!apIt.next().equals(dpIt.next())) {
+
+        for (int i=0; i<ancestorDeckPath.length; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < ancestorDeckPath.length; i++) {
```

I'm leaving these whitespace comments as suggestions but if you actually commit them like this the commit history will be hash and I'll want to squash, better to rebase and fix the original commits so I can rebase+merge and preserve the otherwise-beautifully separated commits

Reply:
I did force push a version directly with the correct spaces (hopefully)

You may have remarked that is what I usually do each time a PR has an error. To leave a beautiful commit history

Reply:
Yes - your commit histories are usually very nice, to the extent they can be (sometimes things are messy because life is messy). Ironically though, this is all fixed now except this one line that has the comment pinned to it :-) - line 546 of libanki/Decks.java